Provide ES6 modules in transpilation context

goldibex opened the pull request:
This change, ripped straight from [plugin-text](https://github.com/systemjs/plugin-text/blob/master/text.js), enables generation of ES6 modules in the case where transpilation is happening. This makes it possible to use plugins like the excellent [systemjs-hot-reloader](https://github.com/capaj/systemjs-hot-reloader) for a much smoother development experience.

goldibex committed:
Provide ES6 modules in transpilation context

This change, ripped straight from [plugin-text](https://github.com/systemjs/plugin-text/blob/master/text.js), enables generation of ES6 modules in the case where transpilation is happening. This makes it possible to use plugins like the excellent [systemjs-hot-reloader](https://github.com/capaj/systemjs-hot-reloader) for a much smoother development experience.

KevCJones commented:
Thanks for this, taking a look now. I don't suppose you've got something out there showing what you the two working together. This is actually something quite useful to me.

goldibex commented:
Unfortunately I don't; I could potentially throw up a Codepen if that
blocks the PR.
Thanks for this plugin by the way. Was banging my head on sass.js before I
saw it.

On Mon, Apr 25, 2016 at 1:52 PM Kevin C Jones <notifications@github.com>
wrote:

> Thanks for this, taking a look now. I don't suppose you've got something
> out there showing what you the two working together. This is actually
> something quite useful to me.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/KevCJones/plugin-scss/pull/9#issuecomment-214517918>
>

KevCJones commented:
I'm happy enough someone else is getting value from it. So no thanks needed.

Running review now and at the moment right now the test ```gulp test:runtime``` doesn't inject so i'm just debugging it now to see what changes are needed (either at the importing css or in the plugin to fix).

goldibex commented:
Hm, I should probably have added a test. Will look after that now.


On Mon, Apr 25, 2016 at 2:16 PM Kevin C Jones <notifications@github.com>
wrote:

> I'm happy enough someone else is getting value from it. So no thanks
> needed.
>
> Running review now and at the moment right now the test gulp test:runtime
> doesn't inject so i'm just debugging it now to see what changes are needed
> (either at the importing css or in the plugin to fix).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/KevCJones/plugin-scss/pull/9#issuecomment-214525364>
>

KevCJones commented:
I can give you these couple of changes then as well to factor in the fix for runtime test.

[Patch in past-bin because i've not quite figured some things out on github :) ](http://pastebin.com/9RxrN3rg)

Unwrapping the css if its a 'default' exists, seemed the least abusive way to solve this issue. The others were just tweaks and lint fixes.

Commit:
Changes per @KevCJones

- fix lint error
- unwrap default if exists in injectcss
- set compact to false in babel settings to mute babel optamisation error >100kb

Commit:
added test for esm export

goldibex commented:
Okay, tests pass as written.

goldibex commented:
The failures appear to be related to CI configuration, not the PR.

KevCJones commented:
Ah yeah sorry I didn't mean Travis, the tests so far are more acceptance
based. Not ideal but I've not come up with a better way yet. Will review
this PR again when I get into work. Many thanks again
On 26 Apr 2016 03:46, "Harry Schmidt" <notifications@github.com> wrote:

> The failures appear to be related to CI configuration, not the PR.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/KevCJones/plugin-scss/pull/9#issuecomment-214591409>
>

KevCJones commented:
Ok i'm happy enough, i need to reconfigure travis a little i think, also one thing i'll do post merge is just tidy up the ```gulp lint```. I didn't stress it in any way but it will fail travis if any eslint errors return.

Many thanks for the effort, helping hands like this makes open source fun.

K

KevCJones commented:
After lint fix - [Travis was happy enough it seems.](https://travis-ci.org/KevCJones/plugin-scss/builds/125774691)



Thanks again
